### PR TITLE
Update log4j, mockito, restassured EDGCOMMON-23 EDGCOMMON-24

### PR DESCRIPTION
**Pull request by adamdickmeiss**
Note the the TestUtils.assertLogMessage implementation is incomplete. This utility was not tested before this PR and does not seem to be used anywhere (only in edge-common).

Tried compiling this branch with edge-patron. No changes needed to edge-patron code, except the pom update to use the snapshot.

### CODE REVIEW

File: src/main/java/org/folio/edge/core/utils/test/TestUtils.java [hunk 1]
```diff
@@ -1,60 +1,132 @@
...
+    Assert.assertNull(t);
 
-      if (t != null) {
-        assertNotNull(argument.getValue().getThrowableInformation());
-        assertEquals(t, argument.getValue().getThrowableInformation().getThrowable());
-      }
-    } finally {
-      logger.removeAppender(appender);
+    if (logLevel != null) {
+      Assert.assertTrue(!appender.events.isEmpty());
```
Comment (craigmcnally) [inline src/main/java/org/folio/edge/core/utils/test/TestUtils.java:104]: Why do we check this here?  What if we only wanted to check that the log level was set w/o ever having logged a message?

File: src/main/java/org/folio/edge/core/utils/test/TestUtils.java [hunk 2]
```diff
@@ -1,60 +1,132 @@
...
+      Level gotLevel = appender.events.get(0).getLevel();
+      Assert.assertTrue(gotLevel == Level.OFF || gotLevel == logLevel);
```
Comment (craigmcnally) [inline src/main/java/org/folio/edge/core/utils/test/TestUtils.java:106]: I would expect this to fail if the log level doesn't match what was provided.  Why the special case for `Level.OFF`?

File: src/main/java/org/folio/edge/core/utils/test/TestUtils.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/core/utils/test/TestUtils.java:104]: If no logging is made, it's not possible to check for log level fired off. But I'm happy to say, that case is true (no failures).

File: src/main/java/org/folio/edge/core/utils/test/TestUtils.java (hunk 2 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/core/utils/test/TestUtils.java:106]: The implementation is BROKEN!! The system always returns Level.OFF. I've spent 4 hours to figure out why.. Open to suggestions.. However, this utility is ONLY used in edge-common.

File: src/main/java/org/folio/edge/core/utils/test/TestUtils.java (hunk 1 above)
Comment (craigmcnally) [reply src/main/java/org/folio/edge/core/utils/test/TestUtils.java:104]: right, got it.

File: src/main/java/org/folio/edge/core/utils/test/TestUtils.java (hunk 2 above)
Comment (craigmcnally) [reply src/main/java/org/folio/edge/core/utils/test/TestUtils.java:106]: interesting.  I don't have much log4j2 experience.  @mreno-EBSCO ?

Comment (mreno-EBSCO) [reply src/main/java/org/folio/edge/core/utils/test/TestUtils.java:106]: I had started to upgrade to log4j2 in my local edge-common repo and ended up removing the ability to assert log messages as I wasn't making much progress. I also was motivated to remove compile time dependencies on test libraries, which end up as required dependencies in the edge modules, so maybe I didn't try hard enough. 😄

Comment (adamdickmeiss) [reply src/main/java/org/folio/edge/core/utils/test/TestUtils.java:106]: How about we just leave it for now, but remove it for version 4.

Alternatively, I'm happy to remove it in this branch. I don't know of ANY modules using that feature from mod-edge.

Comment (craigmcnally) [reply src/main/java/org/folio/edge/core/utils/test/TestUtils.java:106]: It's probably safer to remove it in v4.  That get's my vote.

**Review by craigmcnally — APPROVED** [review]
